app.js

```diff
@@ -1,24 +1,259 @@
+
+app.get('/hashtags', (req, res) => {
+    readFile().then(posts => {
+        let hashtags = new Set();
+
+        for (let post of posts) {
+            for (let i = 0; i < post.hashtags.length; i++) {
+                hashtags.add(post.hashtags[i]);
+            }
+        }
+        let response = Array.from(hashtags);
+        res.json(JSON.stringify(response));
+        res.end();
+    });
+});
+
+app.post('/getPhotoPosts', (req, res) => {
+    let skip = req.query.skip;
+    let top = req.query.top;
+    let filterConfig = req.body;
+
+    fs.readFile(path.join(__dirname, 'public/posts.json'), (err, data) => {
+        if (err) throw err
+        let posts = JSON.parse(data)
+
+        posts = posts.filter(x => !x.deleted)
+
+        if (filterConfig !== undefined) {
+            if (filterConfig.hashtags) {
+                for (let hashtag of filterConfig.hashtags) {
+                    posts = posts.filter(x => {
+                        return this.intersection([hashtag], x.hashtags);
+                    });
+                }
+
+            }
+
+            if (filterConfig.author) {
+                posts = posts.filter(x => x.author.includes(filterConfig.author));
+            }
+
+            if (filterConfig.dateFrom) {
+                posts = posts.filter(x => x.createdAt > filterConfig.dateFrom);
+            }
+
+            if (filterConfig.dateTo) {
+                posts = posts.filter(x => x.createdAt < filterConfig.dateTo);
+            }
+
+        }
+        posts.sort(function (a, b) {
+            return Date.parse(b.createdAt) - Date.parse(a.createdAt);
+        });
+        posts.slice(skip, skip + top);
+
+        res.json(posts);
+        res.end();
+    })
+});
 
-	fs.readFile(filePath, (error, file) => {
-		if (error) {
-			response.writeHead(404);
-			response.write('Page not found', filePath);
- 		} else {
- 			response.writeHead(200);
- 			response.write(file);
- 		}
- 		response.end();
-	});
+app.post('/updatePosts', (req, res) => {
+    let posts = req.body;
+    fs.writeFile(path.join(__dirname, 'public/posts.json'), JSON.stringify(posts), (err) => {
+        console.log(err);
+        res.end();
+    });
 });
 
-server.listen('8081', () => {
-	console.log('Server is running on port 8081');
-});
\ No newline at end of file
+app.post('/addPhotoPost', (req, res) => {
+    let postToAdd = req.body;
+
+    let date = new Date(postToAdd.createdAt);
+    postToAdd.createdAt = date;
+
+    postToAdd.id = Date.now().toString();
+    postToAdd.deleted = false;
+
+    try {
+        if (validatePhotoPost(postToAdd)) {
+            readFile().then(posts => {
+                posts.push(postToAdd);
+                fs.writeFile(path.join(__dirname, 'public/posts.json'), JSON.stringify(posts), () => {
+                    res.end();
+                });
+            })
+        }
+
+    } catch (e) {
+        res.writeHead(500);
+        res.end();
+    }
+});
+
+app.post('/likePhotoPost', async (req, res) => {
```

Comment:
``likePhotoPost`` is not fully completed in postman collection , I can't check it